### PR TITLE
Target interface refactor and directory layout cleanup

### PR DESCRIPTION
**Pull request by derekparker**
This patch does two things, so could be broken up, but since it's disruptive (file moves) I wanted to get it in quickly to prevent blocking other work.

Essentially this patch introduces a new interface `target.Interface` which represents the debuggers target, which currently is a system process but could eventually be a core file, etc. The main goal is to continue to abstract, isolate, and cleanup the `proc` package, whose usage going forward is strictly to a be an extremely low-level package for dealing with and manipulation of system processes.

Additionally, I've moved all scattered top level packages into `pkg/` for consistency.

### CODE REVIEW

File: pkg/target/target.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	PCToLine(uint64) (string, int, *gosym.Func)
+	FirstPCAfterPrologue(fn *gosym.Func, sameline bool) (uint64, error)
+}
+
+// ThreadInfo is an interface for getting information on active threads
+// in the process.
+type ThreadInfo interface {
+	Threads() map[int]*proc.Thread
+	CurrentThread() *proc.Thread
+	SwitchThread(int) error
```
Comment (aarzilli) [inline pkg/target/target.go]: SwitchThread seems more at home in ProcessManipulation

File: pkg/target/target.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+// GoroutineInfo is an interface for getting information on running goroutines.
+type GoroutineInfo interface {
+	GoroutinesInfo() ([]*proc.G, error)
+	SelectedGoroutine() *proc.G
+	SwitchGoroutine(int) error
```
Comment (aarzilli) [inline pkg/target/target.go]: SwitchGoroutine seems more at home in ProcessManipulation, like SwitchThread.

File: pkg/target/target.go (hunk 1 above)
Comment (derekparker) [inline pkg/target/target.go]: Agreed.

File: pkg/target/target.go (hunk 2 above)
Comment (derekparker) [inline pkg/target/target.go]: Agreed.

**Comment by derekparker** [comment]
@aarzilli updated.

**Comment by aarzilli** [comment]
LGTM

**Pull request merged**